cli/azd/pkg/tools/azcli/ad.go

```diff
@@ -367,3 +373,105 @@ func (cli *azCli) createRoleAssignmentsClient(
 
 	return client, nil
 }
+
+// Get role definition id from role assignment within specific scope and principalID
+func (cli *azCli) getUserRoleDefinitionID(
+	ctx context.Context,
+	subscriptionId string,
+	scope string,
+	principalId string,
+) ([]string, error) {
+	client, err := cli.createRoleAssignmentsClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	pager := client.NewListForScopePager(scope,
+		&armauthorization.RoleAssignmentsClientListForScopeOptions{
+			Filter: convert.RefOf(fmt.Sprintf("atScope() and assignedTo('%s')", principalId)),
+		})
+
+	roleDefinitionIds := []string{}
+
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed getting next page of role assignments on scope '%s': %w", scope, err)
+		}
+		for _, v := range page.Value {
+			roleDefinitionIds = append(roleDefinitionIds, *v.Properties.RoleDefinitionID)
+		}
+	}
+
+	if len(roleDefinitionIds) == 0 {
+		return nil, fmt.Errorf("role assignment with scope '%s' was not found", scope)
+	}
+
+	return roleDefinitionIds, nil
+}
+
+// Get role definition name from role definition id
+func (cli *azCli) getUserRoleDefinitionName(
+	ctx context.Context,
+	subscriptionId string,
+	scope string,
+	principalId string,
+) ([]string, error) {
+	roleDefinitionName := []string{}
+	client, err := cli.createRoleDefinitionsClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	roleDefinitionID, err := cli.getUserRoleDefinitionID(ctx, subscriptionId, scope, principalId)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, id := range roleDefinitionID {
+		roleDefinitionResult, err := client.GetByID(ctx, id, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failing to get role definition name from role definition id: %w", err)
+		}
+		roleDefinitionName = append(roleDefinitionName, *roleDefinitionResult.Properties.RoleName)
+	}
+
+	return roleDefinitionName, nil
+}
+
+func (cli *azCli) CheckRoleAssignments(
```

Comment:
It feels to me that this is the wrong place for this method to live.  As a matter of policy, `pipeline configure` wants to ensure that before it starts the current principal has either the "Owner" or "User Access Administrator" policies applied to it, because if it does not, it wants to warn the user that the operation may fail later when it tries to assign roles to the newly created service principal.  That feels like logic that is very tied to the policy of the pipeline config gesture, and so it should live in the pipeline manager.

I would export getUserRoleDefinitionName and keep it on this type, so that the code here in CheckRoleAssignments that will move into `pipeline_manager.go` can call it.